tests/ballerina-test/src/test/resources/test-src/expressions/lambda/iterable/iterable-negative.bal

```diff
@@ -67,3 +67,75 @@ function test7(){
     s.filter(function(person p){});
     _ = s.filter(function(string s)(person){return null;});
 }
+
+struct Employee {
+    int id;
+    string name;
+    float salary;
+}
+
+struct EmployeeIncompatible {
+    float id;
+    string name;
+    float salary;
+}
+
+struct EmployeeSalary {
+    int id;
+    float salary;
+}
+
+struct EmployeeSalaryIncompatible {
+    float id;
+    float salary;
+}
+
+function getSalaryInCompatibleInput(EmployeeIncompatible e) (EmployeeSalary) {
+    EmployeeSalary s = {id: e.id, salary: e.salary};
+    return s;
+}
+
+function getSalaryIncompatibleOutput(Employee e) (EmployeeSalaryIncompatible) {
+    EmployeeSalaryIncompatible s = {id: e.id, salary: e.salary};
+    return s;
+}
+
+function getSalaryIncompatibleInputOutput(EmployeeIncompatible e) (EmployeeSalaryIncompatible) {
+    EmployeeSalaryIncompatible s = {id: e.id, salary: e.salary};
+    return s;
+}
+
+function createTable() (table<Employee>) {
+    table<Employee> dt = {};
+
+    Employee e1 = {id:1, name:"A", salary:100};
+    Employee e2 = {id:2, name:"B", salary:200};
+    Employee e3 = {id:3, name:"C", salary:300};
+
+    dt.add(e1);
+    dt.add(e2);
+    dt.add(e3);
+
+    return dt;
+}
+
+function test8() (table) {
+    table<Employee> dt = createTable();
+
+    table<EmployeeSalary> t = dt.select(getSalaryInCompatibleInput);
+    return t;
+}
+
+function test9() (table) {
+    table<Employee> dt = createTable();
+
+    table<EmployeeSalary> t = dt.select(getSalaryIncompatibleOutput);
+    return t;
+}
+
+function test10() (table) {
+    table<Employee> dt = createTable();
+
+    table<EmployeeSalary> t = dt.select(getSalaryIncompatibleInputOutput);
+    return t;
+}
```

Comment:
Shall we create a new file under table for table-iteration-negative.bal

Reply:
Done